bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/GenericEventConditionHandler.java

```diff
@@ -27,50 +30,69 @@
  *
  * @author Benedikt Niehues - Initial contribution
  * @author Kai Kreuzer - refactored and simplified customized module handling
+ * @author Cody Cutrer - refactored to match configuration and semantics of GenericEventTriggerHandler
  */
 @NonNullByDefault
 public class GenericEventConditionHandler extends BaseConditionModuleHandler {
 
     public static final String MODULETYPE_ID = "core.GenericEventCondition";
 
-    private static final String TOPIC = "topic";
-    private static final String EVENTTYPE = "eventType";
-    private static final String SOURCE = "source";
-    private static final String PAYLOAD = "payload";
+    public static final String CFG_TOPIC = "topic";
+    public static final String CFG_TYPES = "types";
+    public static final String CFG_SOURCE = "source";
+    public static final String CFG_PAYLOAD = "payload";
 
     public final Logger logger = LoggerFactory.getLogger(GenericEventConditionHandler.class);
 
+    private final String source;
+    private final @Nullable TopicGlobEventFilter topicFilter;
+    private final Set<String> types;
+    private final @Nullable Pattern payloadPattern;
+
     public GenericEventConditionHandler(Condition module) {
         super(module);
-    }
 
-    private boolean isConfiguredAndMatches(String keyParam, @Nullable String value) {
-        Object mo = module.getConfiguration().get(keyParam);
-        String configValue = mo != null && mo instanceof String ? (String) mo : null;
-        if (configValue != null) {
-            if (PAYLOAD.equals(keyParam)) {
-                // automatically adding wildcards only for payload matching
-                configValue = configValue.startsWith("*") ? configValue : ".*" + configValue;
-                configValue = configValue.endsWith("*") ? configValue : configValue + ".*";
-            }
-            if (value != null) {
-                return value.matches(configValue);
-            } else {
-                return false;
-            }
+        this.source = (String) module.getConfiguration().get(CFG_SOURCE);
+        String topic = (String) module.getConfiguration().get(CFG_TOPIC);
+        if (!topic.isEmpty()) {
+            topicFilter = new TopicGlobEventFilter(topic);
+        } else {
+            topicFilter = null;
+        }
+        if (module.getConfiguration().get(CFG_TOPIC) != null) {
+            this.types = Set.of(((String) module.getConfiguration().get(CFG_TOPIC)).split(","));
+        } else {
+            this.types = Set.of();
+        }
+        String payload = (String) module.getConfiguration().get(CFG_PAYLOAD);
+        if (!payload.isEmpty()) {
+            payloadPattern = Pattern.compile(payload);
         } else {
-            return true;
+            payloadPattern = null;
         }
     }
 
     @Override
     public boolean isSatisfied(Map<String, Object> inputs) {
         Event event = inputs.get("event") != null ? (Event) inputs.get("event") : null;
```

Comment:
```suggestion
        Event event = inputs.get("event") instanceof Event ? (Event) inputs.get("event") : null;
```

Just to make sure we don't run into a `ClassCastException` here.